src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
+    public String getRandomBall(Lottery ball) {
+        int number = new Random().nextInt(100);
+        return "Ball has number: " + number + " and color: " + ball.getRandomColor();
+    }
```

Comment:
The `Lottery` class should not extend `ColorSupplier`. Inheritance is not the correct relationship here because a `Lottery` is not a type of `ColorSupplier`. Instead, you should create an instance of `ColorSupplier` within the `Lottery` class and use it to get a random color.